### ISSUE TITLE
Force convergence using the LLG method at finite temperature

### ISSUE THREAD

**Issue by XueMaXue** [spirit-code/spirit#643]
Dear all,

I am using llg and applying temperature, and using the DEPONDT solver. I would like to ask why the Maximum torque is always around 0.2? Is it not converging? Do I need to run all the steps?

In addition, I would like to know if Spirit still cannot calculate the bulk structure of hcp?

Best regards,
Xue Ma

**Comment by MSallermann** [comment]
Hi @XueMaXue,

this is just a consequence of the finite temperature noise. The random kicks will always perturb the system away from the actual local energy minimum (where the torque would be exactly zero).
The value it saturates at (0.2 in your case), most likely, increases with increasing temperature.

**Comment by XueMaXue** [comment]
Thank you for your answer. If I apply a magnetic field and DEPONDT solver also using, will there be noise?

**Comment by MSallermann** [comment]
> Thank you for your answer. If I apply a magnetic field and DEPONDT solver also using, will there be noise?

Yes, if `llg_temperature` is set to a finite value, there will be noise.

**Comment by XueMaXue** [comment]
If the temperature is 0, can I use this solver? If I use the DEPONDT solver, I find that there is noise.

**Comment by MSallermann** [comment]
The solver should work just as well at zero temperature. Seeing noise in this situation is definitely not expected!

Would be good to verify that the temperature is indeed zero. I don't know how you are running Spirit, but, with default settings, the `llg_temperature` setting should be printed to the terminal.

If you indeed still find noise, it would be interesting to see your exact simulation setup

**Comment by XueMaXue** [comment]
![截屏2024-11-19 17 03 30](https://github.com/user-attachments/assets/73fa9979-6803-4c41-a245-466a32110c0c)
![截屏2024-11-19 17 02 49](https://github.com/user-attachments/assets/509e8caf-05a4-4be0-9f04-f4c21ca2d2c3)
The LLG parameters and log output are shown in the figure. The Maximum torque will not decrease. However, the output shows that the temperature = 0

**Comment by MSallermann** [comment]
Will look into this later. I can test if I can reproduce what you're seeing, but I'll probably need to know a bit more about the hamiltonian parameters and the geometry of the system.

**Comment by XueMaXue** [comment]
[10T 2.zip](https://github.com/user-attachments/files/17813135/10T.2.zip)
Thank you for your reply. 10T 2.zip is my input file. The initial magnetic moment is too large to upload. You can use a random
